pkg/eval/frame.go

```diff
@@ -210,11 +211,13 @@ func (fm *Frame) errorp(r diag.Ranger, e error) Exception {
 		return nil
 	case Exception:
 		return e
+	case errs.SetReadOnlyVar:
+		ctx := diag.NewContext(fm.srcMeta.Name, fm.srcMeta.Code, r)
+		e.VarName = ctx.RelevantString()
+		return &exception{e, &StackTrace{Head: ctx, Next: fm.traceback}}
```

Comment:
This can be deduplicated with the `default` branch below by introducing another type assertion.